### PR TITLE
Add opt out of converter instructions

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR adds instructions on how to opt out of the converter (supported pending the following PR #28536), as well as instructions + examples on how to opt in manually to some components. Note that the link to infraattributes processor and datadog extension README are dead links pending adding READMEs to these components.

### CODE REVIEW

File: comp/otelcol/converter/README.md [hunk 1]
```diff
@@ -29,3 +29,23 @@ For any prometheus receiver collecting collector health metrics, and sending the
...
+
+### Extensions
+
+Please refer to the following example in order to manually set the `pprof`, `health_check`, `zpages` and `datadog` extensions: [extensions.yaml](examples/extensions.yaml). Please refer to the extensions README.md for additional information about the components:
+- [pprof](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/pprofextension/README.md): Enables collecting collector profiles at a defined endpoint.
+- [health_check](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/healthcheckextension/README.md): Enables an HTTP url that can be probed to check the status of the OpenTelemetry Collector.
+- [zpages](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/zpagesextension/README.md): Enables an extension that serves zPages, an HTTP endpoint that provides live data for debugging different components
+- [datadog](../extension/README.md): Enables otel-agent information to be collected in the datadog-agent flare.
```
Comment (songy23) [inline comp/otelcol/converter/README.md:43]: Wrong link?

File: comp/otelcol/converter/examples/prometheus.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+exporters:
+    datadog:
+      api:
+        key: 12345
+
+service:
+    pipelines:
+        metrics:
+            receivers: [nop, prometheus/user-defined]
```
Comment (songy23) [inline comp/otelcol/converter/examples/prometheus.yaml:18]:
```suggestion
            receivers: [prometheus/user-defined]
```

**Review by songy23 — COMMENTED** [review]
I assume no QA is needed?

File: comp/otelcol/converter/README.md [hunk 3]
```diff
@@ -29,3 +29,23 @@ For any prometheus receiver collecting collector health metrics, and sending the
 `GetProvidedConf` and `GetEnhancedConf` return the string representation of the user provided and autoconfigured conf respectively. Currently, these APIs have two limitations:
 - They do not redact sensitive data
 - They do not provide the effective config (including defaults...etc)
+
+## Opting out of converter
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply comp/otelcol/converter/README.md:33]: added in last commit

File: comp/otelcol/converter/README.md [hunk 4]
```diff
@@ -29,3 +29,23 @@ For any prometheus receiver collecting collector health metrics, and sending the
...
+- [datadog](../extension/README.md): Enables otel-agent information to be collected in the datadog-agent flare.
+
+### Prometheus Receiver
+
+The Prometheus receiver scrapes prometheus endpoints. This is used to collect the collectors internal health metrics, by adding a job that scrapes the service telemetry metrics endpoint (configurable via `service::telemetry::metrics`). Please refer to the following example in order to manually set the prometheus receiver: [prometheus.yaml](examples/prometheus.yaml). Please refer to the receivers [README.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver) for additional information about the component.
+
+### Infra Attributes Processor
+
+The infraattributes processor is used to add infra level tags collected by the datadog-agent to your telemetry data. Please refer to the following example in order to manually set the infraattributes processor: [infraattributes.yaml](examples/infraattributes.yaml). Please refer to the processors [README.md](../otlp/components/processor/infraattributesprocessor/README.md) for additional information about the component.
```
[the remark this answers is not in the document]
Comment (mackjmr) [reply comp/otelcol/converter/README.md:51]: This file is added in this PR, see: https://github.com/DataDog/datadog-agent/blob/79b995c8d809d62784b7351c00f489bdc566fd57/comp/otelcol/converter/examples/infraattributes.yaml

File: comp/otelcol/converter/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (songy23) [reply comp/otelcol/converter/README.md:33]: Haven't seen it - have you pushed the commit?

[the remark this answers is not in the document]
Comment (mackjmr) [reply comp/otelcol/converter/README.md:37]: should be good now 👍

**Review by songy23 — APPROVED** [review]
LGTM

File: comp/otelcol/converter/README.md [hunk 5]
```diff
@@ -24,8 +24,32 @@ If it finds datadogexporters which are not defined in a pipeline with the promet
...
 
 `GetProvidedConf` and `GetEnhancedConf` return the string representation of the user provided and autoconfigured conf respectively. Currently, these APIs have two limitations:
 - They do not redact sensitive data
 - They do not provide the effective config (including defaults...etc)
+
+## Opting out of converter
+
+It is possible to opt out of the converter by setting env var `DD_OTELCOLLECTOR_CONVERTER_ENABLED` or agent config `otelcollector.converter.enabled` to `false` (`true` by default). Please note that by doing so, you are removing functionality including flare collection from otel-agent, health metrics from collector, or infra level tagging on your telemetry data. If you want to opt out of some components, you can disable all and add the components that you require manually:
+
+### Extensions
```
Comment (akt-datadog) [inline comp/otelcol/converter/README.md:41]: Can you also add a note in the infra attrs section saying what that processor does

File: comp/otelcol/converter/README.md [hunk 6]
```diff
@@ -24,8 +24,32 @@ If it finds datadogexporters which are not defined in a pipeline with the promet
...
+
+Please refer to the following example in order to manually set the `pprof`, `health_check`, `zpages` and `datadog` extensions: [extensions.yaml](examples/extensions.yaml). Please refer to the extensions README.md for additional information about the components:
+- [pprof](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/pprofextension/README.md): Enables collecting collector profiles at a defined endpoint.
+- [health_check](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/healthcheckextension/README.md): Enables an HTTP url that can be probed to check the status of the OpenTelemetry Collector.
+- [zpages](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/zpagesextension/README.md): Enables an extension that serves zPages, an HTTP endpoint that provides live data for debugging different components
+- [datadog](../extension/README.md): Enables otel-agent information to be collected in the datadog-agent flare.
+
+### Prometheus Receiver
```
Comment (akt-datadog) [inline comp/otelcol/converter/README.md:49]: how come this receiver and the infra processor have two sections each?

Comment (mackjmr) [reply comp/otelcol/converter/README.md:49]: Are you referring to the fact that prometheus receiver and infraattributes processor are both under `## Autoconfigure logic` and `## Opting out of converter` ? If so, the first explains in which manner the converter will add the section, whereas the latter is the addition of instructions on how to add the component manually.

**Pull request merged**